tests/test_linux_pstore.py

```diff
@@ -1,6 +1,39 @@
+import re
+
+
 def test_pstore_fs(shell):
     """
     Test if the pstore filesystem exists.
     """
 
     shell.run_check("test -d /sys/fs/pstore")
+
+
+def test_kernel_messages(shell, check):
+    """
+    Test if the kernel only logs messages that we expect.
+
+    This test will ignore some harmless messages that can happen during normal operation.
+    """
+
+    expected = set(
+        [
+            "spi_stm32 44009000.spi: failed to request tx dma channel",
+            "spi_stm32 44009000.spi: failed to request rx dma channel",
+            "clk: failed to reparent ethck_k to pll4_p: -22",
+            "stm32-dwmac 5800a000.ethernet switch: Adding VLAN ID 0 is not supported",
+        ]
+    )
+
+    allowed = set(
+        [
+            # The following messages can happen during other tests and are harmless
+            "sd 0:0:0:0: [sda] No Caching mode page found",
+            "sd 0:0:0:0: [sda] Assuming drive cache: write through",
+        ]
+    )
```

Comment:
Sets can also be created using [curly braces](https://docs.python.org/3/tutorial/datastructures.html#sets) making this a bit more compact:

```suggestion
    expected = {
        "spi_stm32 44009000.spi: failed to request tx dma channel",
        "spi_stm32 44009000.spi: failed to request rx dma channel",
        "clk: failed to reparent ethck_k to pll4_p: -22",
        "stm32-dwmac 5800a000.ethernet switch: Adding VLAN ID 0 is not supported",
    }

    allowed = {
        # The following messages can happen during other tests and are harmless
        "sd 0:0:0:0: [sda] No Caching mode page found",
        "sd 0:0:0:0: [sda] Assuming drive cache: write through",
     }
```